activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1510,6 +1510,27 @@ def valid_primary_key_options # :nodoc:
       end
 
       private
+        MAXIMUM_GENERATED_INDEX_NAME_LENGTH = 62
+
+        def generate_index_name(table_name, column)
+          name = "index_#{table_name}_on_#{Array(column) * '_and_'}"
+          return name if name.length <= MAXIMUM_GENERATED_INDEX_NAME_LENGTH
+
+          # Remove table_name
+          name = "index_on_#{Array(column) * '_and_'}"
+          return name if name.length <= MAXIMUM_GENERATED_INDEX_NAME_LENGTH
+
+          # Remove _and_ between columns
+          name = "index_on_#{Array(column) * '_'}"
+          return name if name.length <= MAXIMUM_GENERATED_INDEX_NAME_LENGTH
+
+          # Shorten prefix
+          name = "idx_#{Array(column) * '_'}"
```

Comment:
I'm not convinced about this one, it only saves 2 characters and breaks the convention.